testing/run_tests.py

```diff
@@ -75,19 +75,23 @@ def run_cmd(
   print(f'Running command "{command_string}"')
 
   start_time = time.time()
-  collect_output = forbidden_output or allowed_failure_output
-  stdout_pipe = sys.stdout if not collect_output else subprocess.PIPE
-  stderr_pipe = sys.stderr if not collect_output else subprocess.PIPE
 
   process = subprocess.Popen(
       cmd,
-      stdout=stdout_pipe,
-      stderr=stderr_pipe,
+      stdout=subprocess.PIPE,
+      stderr=subprocess.STDOUT,
       env=env,
       universal_newlines=True,
       **kwargs
   )
-  stdout, stderr = process.communicate()
+  output = ''
+
+  for line in iter(process.stdout.readline, ''):
+    output += line
+    sys.stdout.write(line)
```

Comment:
Previously the test script would only generate output when a test fails. This is nice when diagnosing failures on CI since the succeeding tests add only minimal output to the logs. Am I misreading what this does? Is there some way to make this configurable?

Reply:
I believe you are misremembering how it works.  If `collect_output == true` the output would be printed on line 119.  If `collect_output == false` it would get printed out by virtue of using `stdout=sys.stdout`.